bin/legacy-facts-striper

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'json'
+
+# This list of legacy fact was obtained from the facter source code:
+# grep --no-filename -r ALIASES' =' lib | sed -Ee "s/^ *ALIASES = ('|%w\\[)//" -e "s/('|].freeze)\$//" | tr ' ' '\n' | sort | uniq
+LEGACY_FACTS = %w[
+  architecture
+  augeasversion
+  bios_release_date
+  bios_vendor
+  bios_version
+  blockdevice_.*_model
+  blockdevice_.*_size
+  blockdevice_.*_vendor
+  blockdevices
+  boardassettag
+  boardmanufacturer
+  boardproductname
+  boardserialnumber
+  chassisassettag
+  chassistype
+  domain
+  fqdn
+  gid
+  hardwareisa
+  hardwaremodel
+  hostname
+  id
+  ipaddress
+  ipaddress6
+  lsbdistcodename
+  lsbdistdescription
+  lsbdistid
+  lsbdistrelease
+  lsbmajdistrelease
+  lsbminordistrelease
+  lsbrelease
+  macaddress
+  macosx_buildversion
+  macosx_productname
+  macosx_productversion
+  macosx_productversion_major
+  macosx_productversion_minor
+  manufacturer
+  memoryfree
+  memoryfree_mb
+  memorysize
+  memorysize_mb
+  netmask
+  netmask6
+  network
+  network6
+  operatingsystem
+  operatingsystemmajrelease
+  operatingsystemrelease
+  osfamily
+  physicalprocessorcount
+  processor.*
+  processorcount
+  productname
+  rubyplatform
+  rubysitedir
+  rubyversion
+  scope6
+  selinux
+  selinux_config_mode
+  selinux_config_policy
+  selinux_current_mode
+  selinux_enforced
+  selinux_policyversion
+  serialnumber
+  sp_boot_mode
```

Comment:
okay I've never seen those sp_* facts before.

Reply:
Hehe, me neither!  Looks like sp stand for system_profiler and is something related to MaxOSX :shrug:

Reply:
jep, its macOS stuff... its now collected as `system_profiler` structured fact.